src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -151,55 +196,73 @@ class ExtentPlacementManager {
 
     // XXX: bp might be extended to point to differnt memory (e.g. PMem)
     // according to the allocator.
-    auto bp = ceph::bufferptr(
+    auto alloc_paddr = [this](reclaim_gen_t gen, 
+      data_category_t category, extent_len_t length) 
+      -> alloc_result_t {
+      auto bp = ceph::bufferptr(
       buffer::create_page_aligned(length));
-    bp.zero();
+      bp.zero();
+      paddr_t addr;
+      if (category == data_category_t::DATA) {
+	if (data_writers_by_gen.size() <= gen) {
```

Comment:
I don't follow why need to check `data_writers_by_gen.size() <= gen` here, it should be a bug (with assert failure) if the gen is illegal instead of tolerating it.